src/path.c

```diff
@@ -306,6 +306,23 @@ int git_path_join_unrooted(
 	return 0;
 }
 
+void git_path_squash_slashes(git_buf *path)
+{
+	char *p, *q;
+
+	if (path->size == 0)
+		return;
+
+	for (p = path->ptr, q = path->ptr; (*p = *q); p++, q++) {
```

Comment:
I'm not sure how I feel about the `(*p = *q)`. It is concise, but it also looks like you missed an `=`. Since we have the braces and such, would you be against changing the condition to `*q` and putting the `*p = *q` in the body of the loop?